build/source/engine/read_force.f90

```diff
@@ -193,65 +185,65 @@ subroutine read_force(istep,iHRU,err,message)
+   if(time_data(iline)==imiss)then; err=40; message=trim(message)//"variableMissing[var='"//trim(time_meta(iline)%varname)//"']"; return; endif
   end do
  endif
  ! put data in forcing structure
  do iline=1,size(data_ix)
   !print*,data_ix(iline)
   if (data_ix(iline)<1 .or. data_ix(iline)>ncols) cycle
-  read(cline(data_ix(iline)),*,iostat=err) forc_data%var(iline)
+  read(cline(data_ix(iline)),*,iostat=err) forc_data(iline)
   if(err/=0)then; err=30; message=trim(message)//"ProblemDataRead[var='"//trim(forc_meta(iline)%varname)//"']"; return; endif
-  !print*,trim(forc_meta(iline)%varname),forc_data%var(iline)
+  !print*,trim(forc_meta(iline)%varname),forc_data(iline)
  end do
  ! compute the julian day at the start of the year
- call compjulday(time_data%var(iLookTIME%iyyy),          & ! input  = year
+ call compjulday(time_data(iLookTIME%iyyy),          & ! input  = year
                  1, 1, 1, 1, 0._dp,                      & ! input  = month, day, hour, minute, second
                  startJulDay,err,cmessage)                 ! output = julian day (fraction of day) + error control
  if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
  ! compute the fractional julian day for the current time step
- call compjulday(time_data%var(iLookTIME%iyyy),           & ! input  = year
-                 time_data%var(iLookTIME%im),             & ! input  = month
-                 time_data%var(iLookTIME%id),             & ! input  = day
-                 time_data%var(iLookTIME%ih),             & ! input  = hour
-                 time_data%var(iLookTIME%imin),0._dp,     & ! input  = minute/second
+ call compjulday(time_data(iLookTIME%iyyy),           & ! input  = year
+                 time_data(iLookTIME%im),             & ! input  = month
+                 time_data(iLookTIME%id),             & ! input  = day
+                 time_data(iLookTIME%ih),             & ! input  = hour
+                 time_data(iLookTIME%imin),0._dp,     & ! input  = minute/second
                  currentJulday,err,cmessage)                ! output = julian day (fraction of day) + error control
  if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
  ! compute the time since the start of the year (in fractional days)
  fracJulday = currentJulday - startJulDay
  ! compute time since the reference time (in seconds)
- forc_data%var(iLookFORCE%time) = (currentJulday-refJulday)*secprday
+ forc_data(iLookFORCE%time) = (currentJulday-refJulday)*secprday
  ! compute the number of days in the current year
  yearLength = 365
- if(mod(time_data%var(iLookTIME%iyyy),4) == 0)then
+ if(mod(time_data(iLookTIME%iyyy),4) == 0)then
   yearLength = 366
-  if(mod(time_data%var(iLookTIME%iyyy),100) == 0)then
+  if(mod(time_data(iLookTIME%iyyy),100) == 0)then
    yearLength = 365
-   if(mod(time_data%var(iLookTIME%iyyy),400) == 0)then
+   if(mod(time_data(iLookTIME%iyyy),400) == 0)then
     yearLength = 366
    endif
   endif
  endif
```

Comment:
Move to `daysinyear(year)` function that returns number of days and create an `isleapyear(year)` function that returns `.false.` or `.true.`. We should consider supporting multiple calendars.